itou/www/apply/views/process_views.py

```diff
@@ -22,9 +22,14 @@ def check_waiting_period(approvals_wrapper, job_application):
     This should be an edge case.
     An approval may expire between the time an application is sent and
     the time it is accepted.
-    Only "authorized prescribers" can bypass an approval in waiting period.
+    An approval in waiting period can only be bypassed if:
+    - the prescriber is authorized
+    - or the structure is not a SIAE
     """
-    if approvals_wrapper.has_in_waiting_period and not job_application.is_sent_by_authorized_prescriber:
+    bypass_waiting_period_check = (
```

Comment:
Oui la logique est bonne @vgrange.

Par contre je l'aurais exprimée avec un nom plus intuitif dans un contexte booléen et avec des `and`s uniquement :

```python
cannot_bypass_waiting_period = (
    job_application.to_siae.is_subject_to_eligibility_rules
    and approvals_wrapper.has_in_waiting_period
    and not job_application.is_sent_by_authorized_prescriber
)

if cannot_bypass_waiting_period:
    raise …
```

Ce code est dur a factoriser car tu n'as pas de `job_application` au moment de la soumission alors que tu en as une au moment de l'étude.